File: app/src/main/java/com/tobiasfried/rasa/viewmodel/EntryViewModelFactory.java
package com.tobiasfried.rasa.viewmodel;

import com.google.firebase.firestore.FirebaseFirestore;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class EntryViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final FirebaseFirestore mDb;
    private final String mCollection;
    private final String mBrewId;

    public EntryViewModelFactory(FirebaseFirestore database, String collection, String brewId) {
        mDb = database;
        mCollection = collection;
        mBrewId = brewId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new EntryViewModel(mDb, mCollection, mBrewId);
    }

}

File: app/src/main/java/com/tobiasfried/rasa/entry/EntryViewModelFactory.kt
package com.tobiasfried.rasa.entry

import com.google.firebase.firestore.FirebaseFirestore
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class EntryViewModelFactory(private val mDb: FirebaseFirestore, private val mCollection: String, private val mBrewId: String) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return EntryViewModel(mDb, mCollection, mBrewId) as T
    }

}
